satpy/readers/msi_safe.py

```diff
@@ -93,7 +99,21 @@ def _read_from_file(self, key):
         if key["calibration"] == "reflectance":
             return self._mda.calibrate_to_reflectances(proj, self._channel)
         if key["calibration"] == "radiance":
-            return self._mda.calibrate_to_radiances(proj, self._channel)
+            # The calibration procedure differs for L1B and L1C/L2A data!
+            if self.process_level in ["L1C", "L2A"]:
+                # For higher level data, radiances must be computed from the reflectance.
+                # By default, we use the mean solar angles so that the user does not need to resample,
+                # but the user can also choose to use the solar angles from the tile metadata.
+                # This is on a coarse grid so for most bands must be resampled before use.
+                dq = dict(name="solar_zenith_angle", resolution=key["resolution"])
+                zen = self._tile_mda.get_dataset(dq, dict(xml_tag="Sun_Angles_Grid/Zenith"))
+                tmp_refl = self._mda.calibrate_to_reflectances(proj, self._channel)
+                return self._mda.calibrate_to_radiances(tmp_refl, zen, self._channel)
+            #else:
+                # For L1B the radiances can be directly computed from the digital counts.
+                #return self._mda.calibrate_to_radiances_l1b(proj, self._channel)
```

Comment:
Should this be removed?

Reply:
I'd like to leave that there as a placeholder. Soon the L1b data will be made available to users, and this section of code will then become useful :-)

Reply:
Can we uncomment it then?